code/modules/holiday/easter.dm

```diff
@@ -42,25 +46,41 @@
 	emote_hear = list("hops.")
 	emote_see = list("hops around","bounces up and down")
 	butcher_results = list(/obj/item/food/meat/slab = 1)
-	egg_type = /obj/item/food/egg/loaded
-	food_type = /obj/item/food/grown/carrot
-	eggsleft = 10
-	eggsFertile = FALSE
-	icon_prefix = "rabbit"
-	feedMessages = list("It nibbles happily.","It noms happily.")
-	layMessage = list("hides an egg.","scampers around suspiciously.","begins making a huge racket.","begins shuffling.")
 	pet_bonus = TRUE
 	pet_bonus_emote = "hops around happily!"
 	can_be_held = TRUE
+	density = FALSE
+	speak_chance = 2
+	turns_per_move = 3
+	response_help_continuous = "pets"
+	response_help_simple = "pet"
+	response_disarm_continuous = "gently pushes aside"
+	response_disarm_simple = "gently push aside"
+	response_harm_continuous = "kicks"
+	response_harm_simple = "kick"
+	attack_verb_continuous = "kicks"
+	attack_verb_simple = "kick"
+	pass_flags = PASSTABLE | PASSMOB
+	mob_size = MOB_SIZE_SMALL
+	gold_core_spawnable = FRIENDLY_SPAWN
+	///passed to animal_variety component as the prefix icon.
+	var/icon_prefix = "rabbit"
+	///passed to egg_layer component as how many eggs it starts out as able to lay.
+	var/initial_egg_amount = 10
 
-/mob/living/simple_animal/chicken/rabbit/empty //top hats summon these kinds of rabbits instead of the normal kind
-	eggsleft = 0 //if you want to harvest toys and easter bunny gear from these guys, you're gonna need to feed them carrots first
+/mob/living/simple_animal/rabbit/Initialize()
+	. = ..()
+	AddElement(/datum/element/animal_variety, icon_prefix, pick("brown","black","white"), TRUE)
+	AddComponent(/datum/component/egg_layer, /obj/item/food/egg/loaded, list(/obj/item/food/grown/carrot), list("It nibbles happily.","It noms happily."), list("hides an egg.","scampers around suspiciously.","begins making a huge racket.","begins shuffling."), initial_egg_amount, FALSE)
```

Comment:
Split this into multiple lines with `\`, it's currently hard to read.

Also, it's a flaw in the component that we have to use "It" instead of the pronoun helpers. You should figure out a way around that.